src/settings.rs

```diff
@@ -169,22 +165,17 @@ impl Settings {
         unsafe {
             let res = fluid_settings_getint(self.to_raw(), name_str.as_ptr(), &mut value);
 
-            match res {
-                1 => Some(value),
-                _ => None
-            }
+            fluid_ok_option!(res, value)
         }
     }
 
     pub fn getint_default(&self, name: &str) -> Option<i32> {
+        let mut default_int: i32 = 0;
         let name_str = CString::new(name).unwrap();
         unsafe {
-            let res: i32 = fluid_settings_getint_default(self.to_raw(), name_str.as_ptr());
+            let res = fluid_settings_getint_default(self.to_raw(), name_str.as_ptr(), &mut default_int as *mut i32);
 
-            match res {
-                0 => None,
-                _ => Some(res)
-            }
+            fluid_ok_option!(res, res)
```

Comment:
Shouldn't this be like this?

```suggestion
            fluid_ok_option!(res, default_int)
```

Reply:
Yes, you're right. I've corrected my commit and fixed some further errors with fluidsynth's return values.

Thx for taking a look.